arangod/Aql/IResearchViewNode.cpp

```diff
@@ -713,10 +788,10 @@ std::array<std::unique_ptr<aql::ExecutionBlock> (*)(
   }
 };
 
-inline decltype(executors)::size_type getExecutorIndex(bool sorted, bool ordered, MaterializeType materializeType) {
+inline int getExecutorIndex(bool sorted, bool ordered, MaterializeType materializeType) {
   auto index = static_cast<int>(materializeType) + 3 * static_cast<int>(ordered) + 6 * static_cast<int>(sorted);
-  TRI_ASSERT(static_cast<decltype(executors)::size_type>(index) <= executors.size());
-  return static_cast<decltype(executors)::size_type>(index < 9 ? index : index - 1);
+  TRI_ASSERT(static_cast<std::size_t>(index) <= IRESEARCH_COUNTOF(executors));
```

Comment:
shouldn't it be
```
TRI_ASSERT(index < IRESEARCH_COUNTOF(executors));
```
???

Reply:
No. If index >= 9 then returns index - 1